### PR TITLE
Add composer and golang versions 

### PR DESCRIPTION
**Pull request by TG1999**
and also add support for their gitlab version ranges
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

File: src/univers/version_range.py [hunk 1]
```diff
@@ -973,49 +991,46 @@ def from_native(cls, string):
...
-                )
-            else:
-                comparator, version_constraint = split_req(
-                    constraint_item, vrc.vers_by_native_comparators
-                )
-                constraints.append(
-                    VersionConstraint(
-                        comparator=comparator, version=vrc.version_class(version_constraint)
-                    )
+def from_gitlab_native(scheme, string):
```
Comment (pombredanne) [inline src/univers/version_range.py:994]: Is the scheme exactly that of GitLab? or this is ours and there has been a mapping already done upstream in VulnerableCode?

**Review by pombredanne — APPROVED** [review]
LGTM! I just have one question.

File: src/univers/version_range.py (hunk 1 above)
Comment (TG1999) [reply src/univers/version_range.py:994]: This scheme is exactly from GitLab itself, the tests that I have added here are from GitLab only and not changed at all.

Comment (pombredanne) [reply src/univers/version_range.py:994]: Good, so they use purl types like we do then! ... good.

**Review by pombredanne — APPROVED** [review]

File: src/univers/version_range.py (hunk 1 above)
Comment (TG1999) [reply src/univers/version_range.py:994]: One thing, what do you mean by scheme here? The version range or the package type itself? We map the type from slug in yaml file to package url type, and then use as scheme here

**Review by pombredanne — APPROVED** [review]
LGTM! :+1:

**Pull request merged**